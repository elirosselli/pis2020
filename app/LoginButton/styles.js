import { StyleSheet } from 'react-native';

const buttonStyles = StyleSheet.create({
  buttonContainer: {
    backgroundColor: '#005492',
    width: '80%',
    alignSelf: 'center',
    marginTop: 30,
    padding: 15,
    borderRadius: 10,
    borderWidth: 2,
    borderColor: '#222',
    flexDirection: 'row',
  },
  buttonSeparator: {
    borderRightColor: '#ddd',
    borderRightWidth: 1,
    justifyContent: 'center',
  },
  buttonLogo: {
    flex: 1,
    alignSelf: 'center',
    resizeMode: 'center',
    height: '100%',
    aspectRatio: 1,
    marginRight: 15,
  },
  buttonText: {
    color: '#fff',
    fontWeight: 'bold',
    marginLeft: 15,
  }
});

export default buttonStyles;
