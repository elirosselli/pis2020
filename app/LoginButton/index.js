/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-console */
// istanbul ignore file
import React from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import { login, getToken, getParameters, logout } from 'sdk-gubuy-test';
import styles from './styles';
import LogoAgesicSimple from './images/logoAgesicSimple.png';

const stateLog = 'loggedOut'; //  cambiar a loggedIn una vez loggeado, y de nuevo a loggedOut una vez hecho el log out
const idToken = ''; //  poner idToken obtenido con Postman

const LoginButton = () => {
  const handleButton = async () => {
    const parameters = getParameters();
    if (parameters.code === '' || parameters.code === 'empty')
      await handleLogin();
    else await handleLogout();
  };
  const handleLogin = async () => {
    try {
      const code = await login();
      console.log(`Code: ${code}`);
      const token = await getToken();
      console.log(`Token: ${token}`);
      const parameters = getParameters();
      console.log(parameters);
    } catch (err) {
      console.log(err);
      const parameters = getParameters();
      console.log(parameters);
    }
  };

  const handleLogout = async () => {
    try {
      const redirectUri = await logout();
      console.log(redirectUri);
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <TouchableOpacity style={styles.buttonContainer} onPress={handleButton}>
      <View style={styles.buttonSeparator}>
        <Image source={LogoAgesicSimple} style={styles.buttonLogo} />
      </View>
      <Text style={styles.buttonText}>Login con USUARIO gub.uy</Text>
    </TouchableOpacity>
  );
};

export default LoginButton;
