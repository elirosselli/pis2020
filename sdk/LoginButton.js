/* eslint-disable no-console */
// istanbul ignore file
import React from 'react';
import PropTypes from 'prop-types';
import { Image, Text, TouchableOpacity, View } from 'react-native';

import styles from './styles';
import LogoAgesicSimple from './images/logoAgesicSimple.png';
import { login } from './interfaces';

const LoginButton = ({ sdkIdUClientId }) => {
  const handleLogin = async () => {
    try {
      const code = await login(sdkIdUClientId);
      console.log(code);
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <TouchableOpacity style={styles.buttonContainer} onPress={handleLogin}>
      <View style={styles.buttonSeparator}>
        <Image source={LogoAgesicSimple} style={styles.buttonLogo} />
      </View>
      <Text style={styles.buttonText}>Login con USUARIO gub.uy</Text>
    </TouchableOpacity>
  );
};

LoginButton.propTypes = {
  sdkIdUClientId: PropTypes.string.isRequired,
};

export default LoginButton;
