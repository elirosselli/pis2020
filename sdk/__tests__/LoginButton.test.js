import React from 'react';
import renderer from 'react-test-renderer';
import LoginButton from '../LoginButton';

describe('<LoginButton />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<LoginButton sdkIdUClientId="sdkIdUClientId" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Expect to not log error in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderer.create(<LoginButton sdkIdUClientId="sdkIdUClientId" />).toJSON();
    expect(spy).not.toHaveBeenCalled();
  });
});
