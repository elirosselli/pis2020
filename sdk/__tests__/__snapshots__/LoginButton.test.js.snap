// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#005492",
      "borderColor": "#222",
      "borderRadius": 10,
      "borderWidth": 2,
      "flexDirection": "row",
      "marginTop": 30,
      "opacity": 1,
      "padding": 15,
      "width": "80%",
    }
  }
>
  <View
    style={
      Object {
        "borderRightColor": "#ddd",
        "borderRightWidth": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/logoAgesicSimple.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "aspectRatio": 1,
          "flex": 1,
          "height": "100%",
          "marginRight": 15,
          "resizeMode": "center",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontWeight": "bold",
        "marginLeft": 15,
      }
    }
  >
    Login con USUARIO gub.uy
  </Text>
</View>
`;
